app/src/main/java/com/iti/android/team1/ebuy/ui/category/view/CategoryFragment.kt

```diff
@@ -1,48 +1,159 @@
 
         _binding = FragmentCategoryBinding.inflate(inflater, container, false)
         setHasOptionsMenu(true)
 
-        initRecyclerView()
 
         return binding.root
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initCategoriesRecyclerView()
+        initRecyclerView()
+
+        categoryViewModel.getAllCategories()
+        categoryViewModel.getAllProduct()
+
+        lifecycleScope.launchWhenStarted {
+            categoryViewModel.allCategories.buffer().collect {
+                handleCategoriesResult(it)
+            }
+        }
+
+        lifecycleScope.launchWhenStarted {
+            categoryViewModel.allProducts.buffer().collect {
+                handleProductsResult(it)
+            }
+        }
+
+        binding.fabAccessories.setOnClickListener {
+            onFabClickListener("ACCESSORIES")
+        }
+        binding.fabShoes.setOnClickListener {
+            onFabClickListener("SHOES")
+        }
+        binding.fabTShirt.setOnClickListener {
+            onFabClickListener("T-SHIRTS")
+        }
+    }
+
+    private fun onFabClickListener(productType: String) {
+        defaultProductType = productType
+        categoryViewModel.getAllProductByType(defaultCategoryId, defaultProductType)
+    }
+
+    private fun handleCategoriesResult(result: ResultState<Categories>) {
+        when (result) {
+            is ResultState.Loading -> {}
+            is ResultState.Success -> {
+                Log.i("TAG", "handleCategoriesResult: Success")
+                result.data.list?.let {
+                    categoriesAdapter.setList(it)
+                    binding.catTvName.text = it[0].categoryTitle
+                }
+            }
+            is ResultState.EmptyResult -> {}
+            is ResultState.Error -> {}
+        }
+    }
+
+    private fun handleProductsResult(result: ResultState<Products>) {
+        when (result) {
+            is ResultState.Loading -> {
+                startShimmer()
+            }
+            is ResultState.Success -> {
+                stopShimmer()
+                Log.i("TAG", "handleCategoriesResult: Success")
+                result.data.products?.let { categoryProductsAdapter.setList(it) }
+            }
+            is ResultState.EmptyResult -> {}
+            is ResultState.Error -> {}
+        }
+    }
+
+    private fun initCategoriesRecyclerView() {
+        categoriesAdapter = CategoriesAdapter(::onCategoryClick)
+        binding.categoryRecycler.apply {
+            this.adapter = categoriesAdapter
+            this.layoutManager =
+                LinearLayoutManager(context).apply { this.orientation = RecyclerView.HORIZONTAL }
+            this.setHasFixedSize(true)
+        }
+    }
+
+    private fun onCategoryClick(id: Long, title: String) {
+        binding.catTvName.text = title
+        defaultCategoryId=id
+        categoryViewModel.getAllProduct(id)
+    }
```

Comment:
Is there any suggestion for this function, which is used as an argument to the adapter?